packages/rpc-graphql/src/loaders/account.ts

```diff
@@ -1,47 +1,319 @@
+                        const { commitment, dataSlice, minContextSlot } = orphanedArgs;
+                        const defaultOrphanArgs: AccountLoaderArgsBase = {
+                            commitment,
+                            dataSlice,
+                            encoding: 'base64',
+                            minContextSlot,
+                        };
+                        addFetchByArgsHash(defaultOrphanArgs, {
+                            callback: orphanPromiseCallback,
+                            dataSlice: defaultOrphanArgs.dataSlice ?? null,
+                        });
+                    });
+                }
+            });
+
+            /**
+             * For each set of accounts related to some common args, fetch them in the fewest number
+             * of network requests.
+             */
+            Object.values(accountFetchesByArgsHash).forEach(async ({ args, addresses: addressCallbacks }) => {
+                const addresses = Object.keys(addressCallbacks) as Address[];
+                if (addresses.length === 1) {
+                    const address = addresses[0];
+                    try {
+                        const result = await resolveAccountUsingRpc({ address, ...args });
+                        addressCallbacks[address].callbacks.forEach(({ callback, dataSlice }) => {
+                            callback.resolve(sliceData(result, dataSlice, args.dataSlice));
+                        });
+                    } catch (e) {
+                        addressCallbacks[address].callbacks.forEach(({ callback }) => {
+                            callback.reject(e);
+                        });
+                    }
+                } else {
+                    // Maximum number of addresses per batch
+                    // See https://docs.solana.com/api/http#getmultipleaccounts
+                    const chunkSize = 100;
+                    for (let i = 0; i < addresses.length; i += chunkSize) {
+                        const chunk = addresses.slice(i, i + chunkSize);
+                        try {
+                            const results = await resolveMultipleAccountsUsingRpc({ addresses: chunk, ...args });
+                            chunk.forEach((address, ii) => {
+                                const result = results[ii];
+                                addressCallbacks[address].callbacks.forEach(({ callback, dataSlice }) => {
+                                    callback.resolve(sliceData(result, dataSlice, args.dataSlice));
+                                });
+                            });
+                        } catch (e) {
+                            chunk.forEach(address => {
+                                addressCallbacks[address].callbacks.forEach(({ callback }) => {
+                                    callback.reject(e);
+                                });
+                            });
+                        }
+                    }
```

Comment:
So, one of the problems that you're having so much trouble with your tests and the timers therein is that you're resolving these calls _in series_ in different runloops. You want to split the requests up into batches and fire _all_ the requests off in the same tick.

This is the classic ‘`await` in a loop’ problem. Wherever you notice an `await` in a loop, be suspicious. There's even [a linter](https://eslint.org/docs/latest/rules/no-await-in-loop) for this.

```suggestion
                    await Promise.all(
                        Array.from({ length: Math.ceil(addresses.length / chunkSize) }, async (_, ii) => {
                            const startIndex = ii * chunkSize;
                            const endIndex = startIndex + chunkSize;
                            const chunk = addresses.slice(startIndex, endIndex);
                            try {
                                const results = await resolveMultipleAccountsUsingRpc({
                                    addresses: chunk,
                                    ...args,
                                });
                                chunk.forEach((address, ii) => {
                                    const result = results[ii];
                                    addressCallbacks[address].callbacks.forEach(({ callback, dataSlice }) => {
                                        callback.resolve(sliceData(result, dataSlice, args.dataSlice));
                                    });
                                });
                            } catch (e) {
                                chunk.forEach(address => {
                                    addressCallbacks[address].callbacks.forEach(({ callback }) => {
                                        callback.reject(e);
                                    });
                                });
                            }
                        }),
                    );
```